### DISCUSSION TITLE
How can I increase envelope size more than 1mb?

### DISCUSSION THREAD

**Discussion by karambaq** [getsentry/sentry-python#3046] [Q&A]
I have endpoint which have like 10k requests to database so event object becoming more than 1mb and it didn't even showing on sentry. How can I increase this size?

**Comment by antonpirker** [comment]
The [1MB limit on transactions](https://develop.sentry.dev/sdk/event-payloads/#size-limits) is a hard limit and can not be changed. 

Another thing that might affect you is that one transaction has a limit of 1000 spans per transaction. (In your case a database request is one span and because they all happen in one request to your endpoint end up in the same transaction)

This span limit can be changed like this:
```python
sentry_sdk.init(
    dsn=...,
    _experiments={
        "max_spans": 2000,
    }
```

Another thing you could do is use a `before_send` hook to remove unimportant data from the event to make it smaller. See: https://docs.sentry.io/platforms/python/configuration/filtering/
